lib/Extension/Bundle/Traits/PackageVersionTrait.php

```diff
@@ -27,12 +27,25 @@
  */
 trait PackageVersionTrait
 {
+
     /**
      * Returns the composer package name used to resolve the version
      *
      * @return string
      */
-    abstract protected function getComposerPackageName(): string;
+    public function getComposerPackageName(): string {
+        foreach([__DIR__ . '/../composer.json', __DIR__ . '/composer.json'] as $composerFile) {
```

Comment:
I'm wondering if this is really generic enough, as this assumes the bundle class in a specific directory, right?

Reply:
Right. As explained in my previous comment, it may be in one of two places if you adhere to Symfony's "bundle best practices": in the bundle root, or in `src/`.

So, in my opinion, that should be enough, as this is only a default implementation for lazy developers ;) 
If you have some other bundle structure, you can still implement the method yourself, as before.

Reply:
So with @jdreesen explantation, the solution is okay and the thread is resolved? @brusch :)

Reply:
I was wondering whether we can additionally utilize`\Composer\InstalledVersions::getAllRawData()` - at least for installed packages this would be a safe approach ...

Reply:
Should be something like this, I guess:

> **Warning** 
> I didn't test it!

```php
    public function getComposerPackageName(): string
    {
        foreach(InstalledVersions::getAllRawData() as $installed) {
            foreach ($installed['versions'] as $packageName => $packageInfo) {
                if (!isset($packageInfo['install_path'])) {
                    // It's a replaced or provided (virtual) package
                    continue;
                }

                if (str_starts_with(__DIR__, realpath($packageInfo['install_path']))) {
                    return $packageName;
                }
            }
        }

        return '';
    }
```